modules/shadertools/src/lib/transpile-shader.js

```diff
@@ -12,40 +12,54 @@ export default function transpileShader(source, targetGLSLVersion, isVertex) {
   }
 }
 
+const FS_OUTPUT_REGEX = /^[ \t]*out\s+vec4\s+(\w+)\s*;/m;
+
 function convertVertexShaderTo300(source) {
   return source
-    .replace(/attribute\s+/g, 'in ')
-    .replace(/varying\s+/g, 'out ')
-    .replace(/texture2D\(/g, 'texture(')
-    .replace(/textureCube\(+/g, 'texture(')
-    .replace(/texture2DLodEXT\(/g, 'textureLod(')
-    .replace(/textureCubeLodEXT\(/g, 'textureLod(');
+    .replace(/^(#version\s+(100|300\s+es))?[ \t]*\n/, '#version 300 es\n')
+    .replace(/\battribute\s+/g, 'in ')
+    .replace(/\bvarying\s+/g, 'out ')
+    .replace(/\btexture2D\(/g, 'texture(')
+    .replace(/\btextureCube\(+/g, 'texture(')
+    .replace(/\btexture2DLodEXT\(/g, 'textureLod(')
+    .replace(/\btextureCubeLodEXT\(/g, 'textureLod(');
 }
 
 function convertFragmentShaderTo300(source) {
   return source
-    .replace(/varying\s+/g, 'in ')
-    .replace(/texture2D\(/g, 'texture(')
-    .replace(/textureCube\(/g, 'texture(')
-    .replace(/texture2DLodEXT\(/g, 'textureLod(')
-    .replace(/textureCubeLodEXT\(/g, 'textureLod(');
-
-  // Deal with fragColor
-  // .replace(/gl_fragColor/g, 'fragColor ');
+    .replace(/^(#version\s+(100|300\s+es))?[ \t]*\n/, '#version 300 es\n')
+    .replace(/\bvarying\s+/g, 'in ')
+    .replace(/\btexture2D\(/g, 'texture(')
+    .replace(/\btextureCube\(/g, 'texture(')
+    .replace(/\btexture2DLodEXT\(/g, 'textureLod(')
+    .replace(/\btextureCubeLodEXT\(/g, 'textureLod(');
 }
 
 function convertVertexShaderTo100(source) {
   // /gm - treats each line as a string, so that ^ matches after newlines
   return source
-    .replace(/^in\s+/gm, 'attribute ')
-    .replace(/^out\s+/gm, 'varying ')
-    .replace(/texture\(/g, 'texture2D(');
+    .replace(/^#version\s+300\s+es/, '#version 100')
+    .replace(/^[ \t]*in[ \t]+/gm, 'attribute ')
+    .replace(/^[ \t]*out[ \t]+/gm, 'varying ')
+    .replace(/\btexture\(/g, 'texture2D(')
+    .replace(/\btextureLod\(/g, 'texture2DLodEXT(');
 }
 
 function convertFragmentShaderTo100(source) {
   // /gm - treats each line as a string, so that ^ matches after newlines
-  return source.replace(/^in\s+/gm, 'varying ').replace(/texture\(/g, 'texture2D(');
+  source = source
+    .replace(/^#version\s+300\s+es/, '#version 100')
+    .replace(/^[ \t]*in[ \t]+/gm, 'varying ')
+    .replace(/\btexture\(/g, 'texture2D(')
+    .replace(/\btextureLod\(/g, 'texture2DLodEXT(');
+
+  const outputMatch = source.match(FS_OUTPUT_REGEX);
+  if (outputMatch) {
+    const outputName = outputMatch[1];
+    source = source
+      .replace(FS_OUTPUT_REGEX, '')
+      .replace(new RegExp(`\\b${outputName}\\b`, 'g'), 'gl_FragColor');
```

Comment:
Any reason to not use the inline regex syntax?

Reply:
The interpolation of `outputName`. Is there a way to do that with a literal?